pkg/syncer/admissionhandler/admissionhandler.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admissionhandler
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	stroagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+type (
+	parameterSet map[string]struct{}
+	emptyStruct  struct{}
+)
+
+// Has checks if specified paramName present in the paramSet
+func (paramSet parameterSet) Has(paramName string) bool {
+	_, ok := paramSet[paramName]
+	return ok
+}
+
+var (
+	webhookServerPort     = 8443
+	server                *http.Server
+	unSupportedParameters = parameterSet{
+		common.CSIMigrationParams:                   emptyStruct{},
+		common.DiskFormatMigrationParam:             emptyStruct{},
+		common.HostFailuresToTolerateMigrationParam: emptyStruct{},
+		common.ForceProvisioningMigrationParam:      emptyStruct{},
+		common.CacheReservationMigrationParam:       emptyStruct{},
+		common.DiskstripesMigrationParam:            emptyStruct{},
+		common.ObjectspacereservationMigrationParam: emptyStruct{},
+		common.IopslimitMigrationParam:              emptyStruct{},
+	}
+)
+
+const (
+	volumeExpansionErrorMessage = "AllowVolumeExpansion can not be set to true on the in-tree vSphere StorageClass"
+	migrationParamErrorMessage  = "Invalid StorageClass Parameters. Migration specific parameters should not be used in the StorageClass"
+)
+
+// StartWebhookServer starts the webhook server
+func StartWebhookServer(ctx context.Context) error {
+	log := logger.GetLogger(ctx)
+	certs, err := tls.LoadX509KeyPair("/etc/certs/cert.pem", "/etc/certs/key.pem")
+	if err != nil {
+		log.Errorf("failed to load key pair: %v", err)
+		return err
+	}
+	server = &http.Server{
+		Addr:      fmt.Sprintf(":%v", webhookServerPort),
```

Comment:
Same with `webhookServerPort`, maybe something that can be configurable?

Reply:
let me supply this as well in the vsphere.conf file.